flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/TestBaseUtils.java

```diff
@@ -480,7 +480,8 @@ protected static File asFile(String path) {
 			}
 		}
 		
-		assertEquals("Wrong number of elements result", expectedStrings.length, resultStrings.length);
+		assertEquals(String.format("Wrong number of elements result. Expected: %s. Result: %s.", Arrays.toString(expectedStrings), Arrays.toString(resultStrings)),
```

Comment:
The issue here is that it compares lengths of objects and therefore JUnit only prints compared numbers (say 2 and 0) and not content of arrays.